server/isomorphic-routing/index.js

```diff
@@ -13,17 +13,17 @@ export function serverRouter( expressApp, setUpRoute, section ) {
 			combineMiddlewares(
 				setSectionMiddlewareFactory( section ),
 				...middlewares
-			),
-			serverRender
+			)
 		);
 	};
 }
 
 function combineMiddlewares( ...middlewares ) {
-	return function( req, res, next ) {
+	return function( req, res ) {
 		req.context = getEnhancedContext( req );
-		applyMiddlewares( req.context, middlewares );
-		next();
+		applyMiddlewares( req.context, ...middlewares, () => {
+			serverRender( req, res );
+		} );
```

Comment:
Really great spot, and great fix! So apparently the mw chain was broken here all the time for anything async, since we were calling `next()` synchronously.

I have a minor suggestion to keep `combineMiddlewares` mw-agnostic, which is in https://gist.github.com/ockham/2892ef443766a51ef32c8a7b77a8c2ac. See if it works for you and let me know if you like it :-)